cherab/solps/plot.py

```diff
@@ -0,0 +1,110 @@
+import matplotlib.pyplot as plt
+from matplotlib.collections import PolyCollection
+
+
+def plot_quadrangle_mesh(mesh, solps_data=None, ax=None):
+    """
+    Plots the quadrangle mesh grid geometry to a matplotlib figure.
+
+    If solps_data is provided, it is used to colour the faces of the quadrangles in the mesh.
+    If matplotlib axes are provided the collection is added to them them,
+    otherwise a new figure and axes are created.
+
+    :param mesh: SOLPSMesh object
+    :param solps_data: Data array defined on the SOLPS mesh (optional)
+    :param ax: matplotlib axes (optional)
+    :return: matplotlib axes
+    """
+    if ax is None:
+        _, ax = plt.subplots(constrained_layout=True)
+
+    if solps_data is None:
+        collection_mesh = create_quadrangle_polycollection(mesh, facecolor="none", edgecolor='b', linewidth=0.5)
+    else:
+        collection_mesh = create_quadrangle_polycollection(mesh, solps_data)
+    ax.add_collection(collection_mesh)
+
+    ax = format_matplotlib_axes(ax, mesh)
+    return ax
+
+
+def create_quadrangle_polycollection(mesh, solps_data=None, **collection_kw):
+    """
+    Creates a matplotlib PolyCollection object from the quadrangle mesh.
+
+    If solps_data is provided, it is used to colour the faces of the quadrangles in the mesh.
+
+    :param mesh: SOLPSMesh object
+    :param solps_data: Optional[np.ndarray] - Data array defined on the SOLPS mesh
+    :param collection_kw: Keyword arguments for the PolyCollection
+    
+    :return: matplotlib.collections.PolyCollection
+    """
+    verts = mesh.vertex_coordinates[mesh.quadrangles]
+    collection_mesh = PolyCollection(verts, **collection_kw)
+    if solps_data is not None:
+        collection_mesh.set_array(solps_data[mesh.quadrangle_to_grid_map[:, 0], mesh.quadrangle_to_grid_map[:, 1]])
+    return collection_mesh
+
+
+def format_matplotlib_axes(ax, mesh=None):
```

Comment:
I feel like this is micromanaging users' plots a bit. Fine to have a private function in this module to reduce code duplication, but generally it's good to not be too opinionated about how users might want to format their axes. I would remove the calls to this in the demo scripts and just call the appropriate `ax` methods directly. But you can keep this function here (prefixed with an underscore) for use in `plot_triangle_mesh` and `plot_quadrangle_mesh`.